virtualizarr/readers/kerchunk.py

```diff
@@ -230,6 +231,13 @@ def dataset_from_kerchunk_refs(
     return vds
 
 
+def determine_chunk_grid_shape(zarray):
+    return tuple(
+        math.ceil(length / chunksize)
+        for length, chunksize in zip(zarray.shape, zarray.chunks)
+    )
```

Comment:
is that correct? This would allow partial chunks, which we usually want to avoid

Reply:
I already have a utility function for this

https://github.com/zarr-developers/VirtualiZarr/blob/e6407e01e4dc2536a64c3a418a59ead30bb353bc/virtualizarr/manifests/array_api.py#L295

Reply:
But yeah this does allow the final chunk to be smaller if that's what you mean. There should be checks for that in other places, but tbh there may be bugs lurking there.